### PR TITLE
Button binding

### LINKED ISSUE #35: Possible button mapping
Work in progress
https://docs.google.com/presentation/d/1oJZhHlrYburMSjcasFFbvPNTfOH3J4_1lejOg0Bpqvw/edit#slide=id.g2670428c9e5_1_0

### PR DESCRIPTION
**Pull request by wilsonwatson**
Competition buttons. Ready to test.

Needs a field test before merging.

Also adds LEDs (directly copied from 2023 code). RobotTool is updated to 1.1.1 to allow AddressableLEDs to be outside of IO.

Closes #35, #58, #59.

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -133,22 +139,29 @@ private void configureButtonBindings() {
...
+        }, () -> {
+            climber.setLeftPower(0);
+        }), shooter.shootSpeaker(),
+            () -> OperatorState.getCurrentState() == OperatorState.State.kClimb));
         // shoot note to speaker after being intaked
-        operator.rightTrigger().whileTrue(CommandFactory.shootSpeaker(shooter, intake));
-        // set shooter to amp scoring preset position
-        operator.start().onTrue(elevatorWrist.ampPosition());
+        operator.rightTrigger().whileTrue(new ConditionalCommand(new StartEndCommand(() -> {
+            climber.setLeftPower(SmartDashboard.getNumber("Left Climber Power", 0));
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java:158]: Should this be right climber?

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -133,22 +139,29 @@ private void configureButtonBindings() {
...
+        }, () -> {
+            climber.setLeftPower(0);
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java:160]: Should this be right climber?

Comment (wilsonwatson) [reply src/main/java/frc/robot/RobotContainer.java]: fixed both

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -133,22 +144,29 @@ private void configureButtonBindings() {
...
         // spit note currently in robot through shooter
         operator.x().whileTrue(CommandFactory.spit(shooter, intake));
+        // reset apriltag vision
+        operator.b().onTrue(new InstantCommand(() -> s_Swerve.resetPvInitialization()));
+        // spin up shooter
+        operator.leftTrigger().whileTrue(new ConditionalCommand(new StartEndCommand(() -> {
+            climber.setLeftPower(SmartDashboard.getNumber("Left Climber Power", 0));
+        }, () -> {
+            climber.setLeftPower(0);
+        }), shooter.shootSpeaker(),
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java:159]: Do we want this. This only runs the shooter but never the indexer??

Comment (wilsonwatson) [reply src/main/java/frc/robot/RobotContainer.java:159]: Yes, this is the "spin-up" button. It allows the operator to keep the spinner running for short cycles (like rebounds).

**Comment by legoguy1000** [comment]
Need @owenflatman to push final changes on his laptop before merging

**Review by legoguy1000 — APPROVED** [review]

**Pull request merged**